draft-ietf-tls-cross-sni-resumption.md

```diff
@@ -116,6 +120,22 @@ merely lifts a performance-motivated "SHOULD NOT" recommendation from Section
 4.6.1.  Notably, it still relies on the client ensuring that the server
 certificate is valid for the new SNI at the time of session resumption.
 
+If the original server's assertion regarding supporting cross-name resumption
+turns out to be incorrect, the server receiving a misdirected ticket
+will not be able to decrypt it, and will therefore reject it.  This is secure,
+as session resumption may be safely rejected for any reason; however, such
```

Comment:
The reason that this is secure is that the resumption PSK remains available only to the original server.  If the server is not the original, it will be unable to learn anything about the previous session that was not already public knowledge.

What you are saying is that the protocol recovers from this situation gracefully, which is also useful to say, but it is not what ensures that it is secure.

I would use a new sentence rather than a semicolon here before the "however".

Reply:
Thanks for pointing this out!  "recover gracefully" is what I was looking for here.